uhabits-core/src/jvmTest/java/org/isoron/uhabits/core/ui/screens/habits/list/ListHabitsMenuBehaviorTest.kt

```diff
@@ -0,0 +1,214 @@
+
+    private val orderCaptor: KArgumentCaptor<HabitList.Order> = argumentCaptor()
+
+    private val secondaryOrderCaptor: KArgumentCaptor<HabitList.Order> = argumentCaptor()
+
+    @Throws(Exception::class)
+    override fun setUp() {
+        super.setUp()
+        behavior = ListHabitsMenuBehavior(screen, adapter, prefs, themeSwitcher)
+        clearInvocations(adapter)
+    }
+
+    @Test
+    fun testInitialFilter() {
+        whenever(prefs.showArchived).thenReturn(true)
+        whenever(prefs.showCompleted).thenReturn(true)
+        behavior = ListHabitsMenuBehavior(screen, adapter, prefs, themeSwitcher)
+        verify(adapter).setFilter(
+            matcherCaptor.capture()
+        )
+        verify(adapter).refresh()
+        verifyNoMoreInteractions(adapter)
+        clearInvocations(adapter)
+        assertTrue(matcherCaptor.lastValue.isArchivedAllowed)
+        assertTrue(matcherCaptor.lastValue.isCompletedAllowed)
+        whenever(prefs.showArchived).thenReturn(false)
+        whenever(prefs.showCompleted).thenReturn(false)
+        behavior = ListHabitsMenuBehavior(screen, adapter, prefs, themeSwitcher)
+        verify(adapter).setFilter(
+            matcherCaptor.capture()
+        )
+        verify(adapter).refresh()
+        verifyNoMoreInteractions(adapter)
+        assertFalse(matcherCaptor.lastValue.isArchivedAllowed)
+        assertFalse(matcherCaptor.lastValue.isCompletedAllowed)
+    }
+
+    @Test
+    fun testOnSortByColor() {
+        behavior.onSortByColor()
+        verify(adapter).primaryOrder = orderCaptor.capture()
+        assertThat(
+            orderCaptor.lastValue,
+            equalTo(HabitList.Order.BY_COLOR_ASC)
+        )
+    }
+
+    @Test
+    fun testOnSortManually() {
+        behavior.onSortByManually()
+        verify(adapter).primaryOrder = orderCaptor.capture()
+        assertThat(
+            orderCaptor.lastValue,
+            equalTo(HabitList.Order.BY_POSITION)
+        )
+    }
+
+    @Test
+    fun testOnSortScore() {
+        behavior.onSortByScore()
+        verify(adapter).primaryOrder = orderCaptor.capture()
+        assertThat(
+            orderCaptor.lastValue,
+            equalTo(HabitList.Order.BY_SCORE_DESC)
+        )
+    }
+
+    @Test
+    fun testOnSortName() {
+        behavior.onSortByName()
+        verify(adapter).primaryOrder = orderCaptor.capture()
+        assertThat(
+            orderCaptor.lastValue,
+            equalTo(HabitList.Order.BY_NAME_ASC)
+        )
+    }
+
+    @Test
+    fun testOnSortStatus() {
+        whenever(adapter.primaryOrder).thenReturn(HabitList.Order.BY_NAME_ASC)
+        behavior.onSortByStatus()
+        verify(adapter).primaryOrder = orderCaptor.capture()
+        verify(adapter).setSecondaryOrder(
+            secondaryOrderCaptor.capture()
+        )
+        assertThat(
+            orderCaptor.lastValue,
+            equalTo(HabitList.Order.BY_STATUS_ASC)
+        )
```

Comment:
I think these could be in a single line. There are many other similar cases in this file.